apps/nextjs/src/app/[locale]/manage/users/[userId]/_profile-avatar-form.tsx

```diff
@@ -0,0 +1,170 @@
+
+      mutate(
+        {
+          userId: user.id,
+          image: base64Url,
+        },
+        {
+          async onSuccess() {
+            // Revalidate all as the avatar is used in multiple places
+            await revalidatePathAction("/");
+            showSuccessNotification({
+              message: tManageAvatar(
+                "changeImage.notification.success.message",
+              ),
+            });
+          },
+          onError(error) {
+            if (error.shape?.data.code === "BAD_REQUEST") {
+              showErrorNotification({
+                title: tManageAvatar("changeImage.notification.toLarge.title"),
+                message: tManageAvatar(
+                  "changeImage.notification.toLarge.message",
+                  { size: "256KB" },
+                ),
+              });
+            } else {
+              showErrorNotification({
+                message: tManageAvatar(
+                  "changeImage.notification.error.message",
+                ),
+              });
+            }
+          },
+        },
+      );
+    },
+    [mutate, user.id, tManageAvatar],
+  );
+
+  const handleRemoveAvatar = useCallback(() => {
+    openConfirmModal({
+      title: tManageAvatar("removeImage.label"),
+      children: tManageAvatar("removeImage.confirm"),
+      onConfirm() {
+        mutate(
+          {
+            userId: user.id,
+            image: null,
+          },
+          {
+            async onSuccess() {
+              // Revalidate all as the avatar is used in multiple places
+              await revalidatePathAction("/");
+              showSuccessNotification({
+                message: tManageAvatar(
+                  "removeImage.notification.success.message",
+                ),
+              });
+            },
+            onError() {
+              showErrorNotification({
+                message: tManageAvatar(
+                  "removeImage.notification.error.message",
+                ),
+              });
+            },
+          },
+        );
+      },
+    });
+  }, [mutate, user.id, openConfirmModal, tManageAvatar]);
+
+  return (
+    <Box pos="relative">
+      <Menu
+        opened={opened}
+        keepMounted
+        onChange={toggle}
+        position="bottom-start"
+        withArrow
+      >
+        <Menu.Target>
+          <UnstyledButton onClick={toggle}>
+            <UserAvatar user={user} size={200} />
+            <Button
+              component="div"
+              pos="absolute"
+              bottom={0}
+              left={0}
+              size="compact-md"
+              fw="normal"
+              variant="default"
+              leftSection={<IconPencil size={18} stroke={1.5} />}
+            >
+              {t("common.action.edit")}
+            </Button>
+          </UnstyledButton>
+        </Menu.Target>
+        <Menu.Dropdown>
+          <FileButton
+            onChange={handleAvatarChange}
+            accept="image/png,image/jpeg"
```

Comment:
What about ``WebP``? Do we allow ``GiF``?

Reply:
Added both